golang/tracing.go

```diff
@@ -92,12 +92,17 @@ func (s Span) TracingEnabled() bool { return (s.flags & tracingFlagEnabled) == t
 
 // NewChildSpan begins a new child span in the provided Context
 func (s Span) NewChildSpan() *Span {
-	return &Span{
+	childSpan := &Span{
 		traceID:  s.traceID,
 		parentID: s.spanID,
-		spanID:   uint64(rng.Int63()),
 		flags:    s.flags,
 	}
+	if s.spanID == 0 {
```

Comment:
Ah interesting, thanks for the link. This other page http://itszero.github.io/blog/2014/03/03/introduction-to-twitters-zipkin/ made it seem like traceid = spanid, but you're right, the protocol does not require it. I'll keep the additional constraint since it makes it easy to identify the first span for now.